README.md

```diff
@@ -1 +1,523 @@
+
+## 2.2 IPLD Schema
+
+``` ipldsch
+type SignedWorkflow struct {
+  work Workflow (rename "ipvm/workflow")
+  sig  VarSig
+}
+
+type Workflow struct {
+  v       SemVer
+  meta    {String : Any}   (implicit {})
+  parent  nullable TaskRef (implicit Null)
+  global  Config           (implicit {}) 
+  defauts Config           (implicit {})
+  tasks   UCAN.Invocation
+  catch   nullable &Wasm   (implicit Null)
+}
+
+type TaskRef struct {
+  inv  &Invocation
+  task String -- Label for the task
+} representation tuple
+```
+
+## 2.3 JSON Examples
+
+``` json
+{
+  "ipvm/workflow": {
+    "v": "0.1.0",
+    "meta": {
+      "tags": ["fission", "bacalhau", "dag-house"]
+    },
+    "global": {
+      "time": [10, "minutes"],
+    },
+    "defaults": {
+      "gas": 1000,
+      "memory": [10, "mega", "bytes"]
+    },
+    "catch": "bafkreifsaaztjgknuha7tju6sugvrlbiwbyx5jf2pky2yxx5ifrpjscyhe",
+    "tasks": {
+      "v": "0.1.0",
+      "nnc": "02468",
+      "prf": [ 
+        {"/": "bafkreie2cyfsaqv5jjy2gadr7mmupmearkvcg7llybfdd7b6fvzzmhazuy"},
+        {"/": "bafkreibbz5pksvfjyima4x4mduqpmvql2l4gh5afaj4ktmw6rwompxynx4"}
+      ],
+      "run": {
+        "notify-bob": {
+          "with": "mailto://alice@example.com",
+          "do": "msg/send",
+          "inputs": [
+            {
+              "to": "bob@example.com",
+              "subject": "DNSLink for example.com",
+              "body": "Hello Bob!"
+            }
+          ],
+          "meta": {
+            "ipvm/config": {
+              "time": [30, "minutes"],
+              "secret": true
+            }
+          }
+        },
+        "log-as-done": {
+          "with": "https://example.com/report"
+          "do": "crud/update"
+          "inputs": {
+            "from": "mailto://alice@exmaple.com",
+            "to": ["bob@exmaple.com"],
+            "event": "email-notification",
+            "value": {"ucan/promise": ["/", "notify-bob"]}
+          }
+        }
+      }
+    }
+  },
+  "sig": {"/": {"bytes:": "5vNn4--uTeGk_vayyPuNTYJ71Yr2nWkc6AkTv1QPWSgetpsu8SHegWoDakPVTdxkWb6nhVKAz6JdpgnjABppC7"}}
+}
+```
+
+# 3 Configuration
+
+The IPVM configuration struct defines secrecy, quotas, and verification strategy:
+
+| Field    | Type              | Description                             | Required | Default                  |
+|----------|-------------------|-----------------------------------------|----------|--------------------------|
+| `secret` | `Boolean or null` | Whether the output is unsafe to publish | No       | `null`                   |
+| `check`  | `Verification`    | Verification strategy                   | No       | `"attestation"`          |
+| `time`   | `TimeInterval`    | Timeout                                 | No       | `[5, "minutes"]`         |
+| `memory` | `InfoSize`        | Memory limit                            | No       | `[100, "kilo", "bytes"]` |
```

Comment:
Q: on infosizes, would it be better to specify units like `kb`, `ms`, as that can be enumerated upon?

Reply:
Haha everyone hates it when I do this, but I keep trying in every project 😆 

> can be enumerated upon?

It is enumerated! Just composable, so we don't need `kB`, `MB`, TB`, `PB`, `ns`, `us`, `ms`, etc. I guess it's not a huge deal either way. We could also set these fields to `string`.

If you still dislike it I can pull it out.

Reply:
Yeah. no biggie. I've been used to systems like prom, etc, which have:

```
Units supported: B, KB, MB, GB, TB, PB, EB. Ex: "512MB". Based on powers-of-2, so 1KB is 1024B
```

So, are people more used to that, maybe?

Reply:
You're right — these kinds of libraries are very common and pretty much what people expect. Ugh, stringly typed things that I have to parse a bunch of different ways (whitespace etc) is gross 😆 

100% of the time that I've done this someone has mentioned it, so clearly it's against the principle of least surprise. Will revert!

Reply:
I love it, please keep it! 😅